Code/GraphMol/test-valgrind.cpp

```diff
@@ -0,0 +1,64 @@
+// $Id$
+//
+//  Copyright (C) 2001-2008 Greg Landrum and Rational Discovery LLC
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/MonomerInfo.h>
+#include <GraphMol/RDKitQueries.h>
+#include <RDGeneral/types.h>
+#include <RDGeneral/RDLog.h>
+//#include <boost/log/functions.hpp>
+#include <GraphMol/FileParsers/FileParsers.h>
+#include <GraphMol/FileParsers/MolWriters.h>
+#include <GraphMol/SmilesParse/SmilesParse.h>
+#include <GraphMol/SmilesParse/SmilesWrite.h>
+#include <sstream>
+#include <iostream>
+#include <boost/random/uniform_int.hpp>
+
+using namespace std;
+using namespace RDKit;
+
+// memory tests for valgrind
+void testRemoveAtomBond(RWMol &m, int atomidx, int bondidx)
+{
+  if (atomidx>=0) m.removeAtom(atomidx);
+  if (bondidx>=0) m.removeAtom(bondidx);
+}
+
+void testRemovals(RWMol m) {
+  boost::random::mt19937 rng;   
```

Comment:
using the RNG to pick which atoms to remove seems a bit heavy handed. How about just picking a few values?